enclave/src/main/java/org/hyperledger/besu/enclave/Enclave.java

```diff
@@ -26,101 +28,140 @@
 
 import java.io.IOException;
 import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 import com.fasterxml.jackson.core.JsonParseException;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import io.netty.util.AsciiString;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
 
 public class Enclave {
-  private static final Logger LOG = LogManager.getLogger();
+
   private static final ObjectMapper objectMapper = new ObjectMapper();
-  private static final MediaType JSON = MediaType.parse("application/json");
-  private static final MediaType ORION = MediaType.get("application/vnd.orion.v1+json");
+  private static final AsciiString JSON = APPLICATION_JSON;
+  private static final AsciiString ORION = AsciiString.of("application/vnd.orion.v1+json");
 
-  private final URI enclaveUri;
-  private final OkHttpClient client;
+  private static final long REQUEST_TIMEOUT_MS = 500L;
+
+  private final Vertx vertx;
+  private final HttpClient client;
 
   public Enclave(final URI enclaveUri) {
-    this.enclaveUri = enclaveUri;
-    this.client = new OkHttpClient();
+    final HttpClientOptions clientOptions = new HttpClientOptions();
+    clientOptions.setDefaultHost(enclaveUri.getHost());
+    if (enclaveUri.getPort() != -1) {
+      clientOptions.setDefaultPort(enclaveUri.getPort());
+    }
+
+    vertx = Vertx.vertx();
```

Comment:
Thinking it would be better to not create a new vertx for each Enclave. Also if we do create a new Vertx do need to add a shutdown hook so it is shutdown properly?

Reply:
Done - but has required significant refactor